client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,228 @@
-import React from "react";
-import { useRouter } from "next/router";
-import { useAppContext } from "../../context";
+import { useRouter } from 'next/router'
+import React from 'react'
+import styled from 'styled-components'
+import { useAppContext } from '../../context'
+import { desktopFontSize, device, fontWeight, PageTitle } from '../../styles/'
+import { Club } from '../../context/'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
+import { BasicEvent } from '../../context/Base'
+import moment from 'moment'
+
+interface BannerProps {
+  backgroundImageUrl: string
+}
+
+const Banner = styled.div`
+  background-image: url(${(props: BannerProps) => props.backgroundImageUrl || ''});
+  background-size: cover;
+  height: 400px;
+`
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin: 0 auto 100px;
+  width: 65%;
+`
 
 export default function ClubDetail() {
-  const { clubs } = useAppContext();
-  const router = useRouter();
+  const { clubs } = useAppContext()
+  const router = useRouter()
+
+  const { id } = router.query
+  if (typeof id !== 'string') return null
+
+  const club = clubs.get(parseInt(id))
+
+  return (
+    <div>
+      <Banner backgroundImageUrl={club.backgroundImageURL} />
+      <Container>
+        <ClubInfo club={club} />
+        <EventsHostedByClub events={club.events} />
+      </Container>
+    </div>
+  )
+}
+
+const IconContainer = styled.div`
+  margin-top: 30px;
+`
+
+const Logo = styled.img`
+  border-radius: 100%;
+  display: block;
+  margin-top: 20px;
+  width: 100px;
+`
+
+const ClubName = styled(PageTitle)`
+  margin-top: 30px;
+`
+
+const Links = styled.div`
+  display: flex;
+  margin-top: 20px;
+`
+
+const SocialLink = styled.div`
+  text-decoration: underline;
+`
+
+const ClubMetaData = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+  margin-top: 30px;
+`
+
+const MemberCount = styled.div`
+  font-style: italic;
+  font-weight: ${fontWeight.bold};
+`
+
+const Categories = styled.div`
+  display: flex;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  font-size: 12px;
+  margin-left: 5px;
+
+  padding: 5px 10px;
+  @media ${device.mobileS} {
+    padding: 2px 5px;
+  }
```

Comment:
Stumbled into this [thing](https://www.w3schools.com/css/css3_mediaqueries.asp) about responsive web design which says the default should be the mobile style

```suggestion
  padding: 2px 5px;
  @media not all and ${device.mobileS} {
    padding: 5px 10px;
  }
```

Reply:
I'm against this approach, I think media queries applied to mobile make sense. Also w3schools shouldn't be the place where we go to for our source of truth.

Reply:
@xujustinj thoughts?

Reply:
Still a blog, but here's a more in-depth argument https://www.mightyminnow.com/2013/11/what-is-mobile-first-css-and-why-does-it-rock/

So the reason we would put the mobile style first is because that way it renders before the query's modifications. Especially on slower phones, this means there's no [flash of unstyled content](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) that temporarily ruins the experience

Reply:
Changed CSS to be mobile first.